### ISSUE TITLE
Channel closed unexpectedly

### ISSUE THREAD

**Issue by mojoaxel** [lightningnetwork/lnd#1499]
### Background

A channel between my **lnd** node and a **c-lightning** node closed unexpectedly:

```
2018-07-03 07:05:58.025 [INF] HSWC: HTLC manager for ChannelPoint(a6724a8466031fcf165a151ded8fa1de087edbcf796337e62d3ae46a29506d7e:1) started, bandwidth=0 mSAT
2018-07-03 07:05:58.025 [INF] HSWC: Attempting to re-resynchronize ChannelPoint(a6724a8466031fcf165a151ded8fa1de087edbcf796337e62d3ae46a29506d7e:1)
2018-07-03 07:05:58.025 [INF] CNCT: Attempting to update ContractSignals for ChannelPoint(a6724a8466031fcf165a151ded8fa1de087edbcf796337e62d3ae46a29506d7e:1)
2018-07-03 07:05:58.038 [ERR] DISC: unable to reply to peer query: unknown message: *lnwire.ReplyShortChanIDsEnd
2018-07-03 07:05:58.038 [ERR] DISC: unable to reply to peer query: unknown message: *lnwire.ReplyShortChanIDsEnd
2018-07-03 07:05:58.039 [ERR] DISC: unable to reply to peer query: unknown message: *lnwire.ReplyShortChanIDsEnd
2018-07-03 07:05:58.039 [ERR] DISC: unable to reply to peer query: unknown message: *lnwire.ReplyShortChanIDsEnd
2018-07-03 07:05:58.039 [ERR] DISC: unable to reply to peer query: unknown message: *lnwire.ReplyShortChanIDsEnd
2018-07-03 07:05:58.046 [WRN] HSWC: ChannelPoint(a6724a8466031fcf165a151ded8fa1de087edbcf796337e62d3ae46a29506d7e:1): received unknown message of type *lnwire.ChannelReestablish
2018-07-03 07:05:58.062 [INF] HSWC: ChannelLink(a6724a8466031fcf165a151ded8fa1de087edbcf796337e62d3ae46a29506d7e:1) has exited
2018-07-03 07:05:58.109 [WRN] DISC: unable to apply gossip filter for peer=024a8228d764091fce2ed67e1a7404f83e38ea3c7cb42030a2789e73cf3b341365: edge not found
2018-07-03 07:05:58.110 [INF] DISC: Received new remote channel announcement: (*lnwire.AnnounceSignatures)(0xc4211aec60)({
 ChannelID: (lnwire.ChannelID) (len=32 cap=32) 7e6d50296ae43a2de6376379cfdb7e08dea18fed1d155a16cf1f0366844a72a7,
 ShortChannelID: (lnwire.ShortChannelID) 530240:757:1,
 NodeSignature: (lnwire.Sig) (len=64 cap=64) {
  00000000  ce 4e 9c e1 6e cb 23 27  80 96 26 4d c6 98 ba f8  |.N..n.#'..&M....|
  00000010  c0 3f 3d 75 d1 1f 31 a1  e1 f7 33 7c 34 12 db 22  |.?=u..1...3|4.."|
  00000020  14 a7 93 6e 68 b4 43 5a  af 89 7a a7 ac 3e c4 83  |...nh.CZ..z..>..|
  00000030  da 15 2c 38 ca 03 6b 6f  c8 54 2c 4d 97 af c5 99  |..,8..ko.T,M....|
 },
 BitcoinSignature: (lnwire.Sig) (len=64 cap=64) {
  00000000  d0 42 00 fc 06 85 61 83  0e 7e e1 83 9b 53 2d 3d  |.B....a..~...S-=|
  00000010  00 ec 40 3a 96 f3 eb 81  00 c3 94 59 a7 f5 c6 0a  |..@:.......Y....|
  00000020  71 a6 ed 57 2c f9 a9 c4  be 32 7e 8b af fd 46 8d  |q..W,....2~...F.|
  00000030  d7 64 57 78 dd 7b d4 93  bb cb 9e b6 b5 20 a3 f4  |.dWx.{....... ..|
 }
})

2018-07-03 07:05:58.112 [INF] DISC: gossipSyncer(024a8228d764091fce2ed67e1a7404f83e38ea3c7cb42030a2789e73cf3b341365): buffering chan range reply of size=265
2018-07-03 07:05:58.112 [INF] DISC: gossipSyncer(024a8228d764091fce2ed67e1a7404f83e38ea3c7cb42030a2789e73cf3b341365): filtering through 265 chans
2018-07-03 07:05:58.113 [INF] DISC: gossipSyncer(024a8228d764091fce2ed67e1a7404f83e38ea3c7cb42030a2789e73cf3b341365): starting query for 6 new chans
2018-07-03 07:05:58.113 [INF] DISC: gossipSyncer(024a8228d764091fce2ed67e1a7404f83e38ea3c7cb42030a2789e73cf3b341365): querying for 6 new channels
2018-07-03 07:05:58.133 [ERR] DISC: unable to reply to peer query: unknown message: *lnwire.ReplyShortChanIDsEnd
2018-07-03 07:05:58.133 [ERR] DISC: unable to reply to peer query: unknown message: *lnwire.ReplyShortChanIDsEnd
2018-07-03 07:05:58.134 [ERR] DISC: unable to reply to peer query: unknown message: *lnwire.ReplyShortChanIDsEnd
2018-07-03 07:05:58.134 [ERR] DISC: unable to reply to peer query: unknown message: *lnwire.ReplyShortChanIDsEnd
2018-07-03 07:05:58.135 [ERR] DISC: unable to reply to peer query: unknown message: *lnwire.ReplyShortChanIDsEnd
2018-07-03 07:05:58.135 [ERR] DISC: unable to reply to peer query: unknown message: *lnwire.ReplyShortChanIDsEnd
2018-07-03 07:05:58.135 [ERR] DISC: unable to reply to peer query: unknown message: *lnwire.ReplyShortChanIDsEnd
2018-07-03 07:05:58.135 [ERR] DISC: unable to reply to peer query: unknown message: *lnwire.ReplyShortChanIDsEnd
2018-07-03 07:05:58.135 [ERR] DISC: unable to reply to peer query: unknown message: *lnwire.ReplyShortChanIDsEnd
2018-07-03 07:05:58.135 [ERR] DISC: unable to reply to peer query: unknown message: *lnwire.ReplyShortChanIDsEnd
2018-07-03 07:05:58.143 [INF] DISC: gossipSyncer(024a8228d764091fce2ed67e1a7404f83e38ea3c7cb42030a2789e73cf3b341365): no more chans to query
```

Here the corresponding the c-lightning log:

```        
2018-07-03T07:05:54.771Z lightning_gossipd(1690): peer 02ab16f790c7727eac2048771cbe95cbcc1aa93931558c0fd87c06d5c389475a7b: Bad concurrent query_short_channel_ids
2018-07-03T07:05:54.771Z lightning_gossipd(1690): peer 02ab16f790c7727eac2048771cbe95cbcc1aa93931558c0fd87c06d5c389475a7b: Bad concurrent query_short_channel_ids
2018-07-03T07:05:54.772Z lightningd(1681): lightning_channeld-02ab16f790c7727eac2048771cbe95cbcc1aa93931558c0fd87c06d5c389475a7b chan #49: Gossipd told us to send error
2018-07-03T07:05:54.772Z lightningd(1681): lightning_channeld-02ab16f790c7727eac2048771cbe95cbcc1aa93931558c0fd87c06d5c389475a7b chan #49: Peer connection lost
2018-07-03T07:05:54.772Z lightningd(1681): 02ab16f790c7727eac2048771cbe95cbcc1aa93931558c0fd87c06d5c389475a7b chan #49: Peer transient failure in CHANNELD_NORMAL: lightning_channeld: Owning subdaemon lightning_channeld died (62208)
2018-07-03T07:05:54.773Z lightning_gossipd(1690): Forgetting remote peer 02ab16f790c7727eac2048771cbe95cbcc1aa93931558c0fd87c06d5c389475a7b
2018-07-03T07:05:54.788Z lightning_gossipd(1690): Connected out for 02ab16f790c7727eac2048771cbe95cbcc1aa93931558c0fd87c06d5c389475a7b
2018-07-03T07:05:54.819Z lightning_gossipd(1690): peer 02ab16f790c7727eac2048771cbe95cbcc1aa93931558c0fd87c06d5c389475a7b now remote
2018-07-03T07:05:54.820Z lightningd(1681): 02ab16f790c7727eac2048771cbe95cbcc1aa93931558c0fd87c06d5c389475a7b chan #49: Peer has reconnected, state CHANNELD_NORMAL
2018-07-03T07:05:54.825Z lightningd(1681): lightning_channeld-02ab16f790c7727eac2048771cbe95cbcc1aa93931558c0fd87c06d5c389475a7b chan #49: pid 23625, msgfd 25
2018-07-03T07:05:54.826Z lightningd(1681): 02ab16f790c7727eac2048771cbe95cbcc1aa93931558c0fd87c06d5c389475a7b chan #49: Already have funding locked in (and ready to announce)
2018-07-03T07:05:54.853Z lightningd(1681): lightning_channeld-02ab16f790c7727eac2048771cbe95cbcc1aa93931558c0fd87c06d5c389475a7b chan #49: init LOCAL: remote_per_commit = 0329d6fcc76bd41da6c9e9f948362c3ee48acb559a17ebf61025b740f6094dbe50, old_remote_per_commit = 03e62c163e77b69e052126ae7edce0465667403a51f396806d6c2f66d14ac47b80 next_idx_local = 13 next_idx_remote = 13 revocations_received = 12 feerates 1570/1570 (range 253-15700)
```

### Your environment

* `Linux 4.4.0-121-generic Ubuntu x86_64 GNU/Linux`
* `lnd 0.4.2-beta commit=9205720bea931d73e55a886f94d21a759180ff2b`
* `Bitcoin Core Daemon version v0.16.0.0-g4b4d7eb255`

**Comment by Roasbeef** [comment]
Can you try to grep the logs for more instances of `024a8228d764091fce2ed67e1a7404f83e38ea3c7cb42030a2789e73cf3b341365`? Then we can see why the channel was closed out.

**Comment by mojoaxel** [comment]
```
cat ~/.lnd/logs/bitcoin/mainnet/lnd.log | grep 024a8228d764091fce2ed67e1a7404f83e38ea3c7cb42030a2789e73cf3b341365
2018-07-03 21:59:11.166 [INF] CRTR: New channel discovered! Link connects 024a8228d764091fce2ed67e1a7404f83e38ea3c7cb42030a2789e73cf3b341365 and 02d0f53a403a0b16efdec1983a082ab7645e35858f6a855871e489c9992748a059 with ChannelPoint(431bfac6bd0ab96bd3d2ec96607585b98e2823b023c6d6b2cb48e643a82cf2cd:1): chan_id=573691082632069121, capacity=0.01337 BTC
2018-07-03 21:59:11.348 [INF] CRTR: New channel discovered! Link connects 024a8228d764091fce2ed67e1a7404f83e38ea3c7cb42030a2789e73cf3b341365 and 02a456b816697c6b8407ee2176475d74b790a96143bc30e896e8bf7a8852a8b637 with ChannelPoint(3f96ffdc2629c1d4c821f2e4278b3203fbde62278c2e488ed717a3486eff7fdb:0): chan_id=575078666274340864, capacity=0.005 BTC
2018-07-03 22:05:13.749 [INF] CRTR: New channel discovered! Link connects 024a8228d764091fce2ed67e1a7404f83e38ea3c7cb42030a2789e73cf3b341365 and 035f813598df61688b7eff80d546b2ba200bb00c1726bf32f44b692cc98a3580cd with ChannelPoint(9155ecae4a63649392dbb57286f773ad98d4165aff29fde76957e5b45a56c3d7:1): chan_id=573499767611588609, capacity=0.01337 BTC
2018-07-04 07:05:57.926 [INF] DISC: Removing gossipSyncer for peer=024a8228d764091fce2ed67e1a7404f83e38ea3c7cb42030a2789e73cf3b341365
2018-07-04 07:05:57.957 [INF] SRVR: finalizing connection to 024a8228d764091fce2ed67e1a7404f83e38ea3c7cb42030a2789e73cf3b341365, inbound=true
2018-07-04 07:05:57.981 [INF] PEER: NodeKey(024a8228d764091fce2ed67e1a7404f83e38ea3c7cb42030a2789e73cf3b341365) loading ChannelPoint(0bbb2e3d4b9974c529d6f41f4a66912b2b26024e375bab494ebe704225f9c7d0:0)
2018-07-04 07:05:58.031 [INF] SRVR: Negotiated chan series queries with 024a8228d764091fce2ed67e1a7404f83e38ea3c7cb42030a2789e73cf3b341365
2018-07-04 07:05:58.031 [INF] DISC: Creating new gossipSyncer for peer=024a8228d764091fce2ed67e1a7404f83e38ea3c7cb42030a2789e73cf3b341365
2018-07-04 07:05:58.031 [INF] DISC: gossipSyncer(024a8228d764091fce2ed67e1a7404f83e38ea3c7cb42030a2789e73cf3b341365): requesting new chans from height=530262 and 4294437033 blocks after
2018-07-04 07:05:58.172 [WRN] DISC: unable to apply gossip filter for peer=024a8228d764091fce2ed67e1a7404f83e38ea3c7cb42030a2789e73cf3b341365: edge not found
2018-07-04 07:05:58.176 [INF] DISC: gossipSyncer(024a8228d764091fce2ed67e1a7404f83e38ea3c7cb42030a2789e73cf3b341365): buffering chan range reply of size=162
2018-07-04 07:05:58.179 [INF] DISC: gossipSyncer(024a8228d764091fce2ed67e1a7404f83e38ea3c7cb42030a2789e73cf3b341365): filtering through 162 chans
2018-07-04 07:05:58.179 [INF] DISC: gossipSyncer(024a8228d764091fce2ed67e1a7404f83e38ea3c7cb42030a2789e73cf3b341365): starting query for 3 new chans
2018-07-04 07:05:58.180 [INF] DISC: gossipSyncer(024a8228d764091fce2ed67e1a7404f83e38ea3c7cb42030a2789e73cf3b341365): querying for 3 new channels
2018-07-04 07:05:58.209 [INF] DISC: gossipSyncer(024a8228d764091fce2ed67e1a7404f83e38ea3c7cb42030a2789e73cf3b341365): no more chans to query
2018-07-04 07:05:58.210 [INF] DISC: gossipSyncer(024a8228d764091fce2ed67e1a7404f83e38ea3c7cb42030a2789e73cf3b341365): applying gossipFilter(start=2018-07-04 06:05:58.210431738 +0000 UTC m=+126805.720894812)
```

**Comment by wpaulino** [comment]
@mojoaxel could you pinpoint the channel point of the force closed channel and grep for that in the logs instead?

**Comment by Roasbeef** [comment]
Closing this as the funds have been swept out on chain: https://www.smartbit.com.au/tx/5b7d1a3086fa0739b89d8d1b6b689a4c914d4ef89255a816c47270304ce44414.

**Comment by martinbehrens** [comment]
@Roasbeef Could you elaborate a bit on why that became a non issue due to the funds being swept out on chain? I guess @mojoaxel would be interested as well.

**Comment by mojoaxel** [comment]
We still have the problem that channels get closed unexpectedly. 
@wpaulino I think you guys should have a look! Is there any way wen can help debugging this?

**Comment by halseth** [comment]
@mojoaxel If you can reproduce this it would be helpful if you could post more extensive logs. That way we could figure out if it's `lnd` or `c-lightning` that is closing the channel :)

**Comment by mojoaxel** [comment]
@halseth [I uploaded my full lnd log](https://gist.github.com/mojoaxel/d6f43947731600d9d1bba299e45551ed). I just removed the most verbose messages but it's still a lot. In the given timeframe we had the same problem twice.

**Comment by martinbehrens** [comment]
@mojoaxel @halseth Added the [c-lightning](https://gist.github.com/martinbehrens/d3bd8b91654c58534fcd6dde79c7ff85) debug log as well.

**Comment by halseth** [comment]
I see some strange link behaviour during link startup:
```
2018-07-05 05:24:01.643 [INF] HSWC: ChannelLink(e8d984eaed827b9ea264aea49c4d6232dfbd76ec4fb218a0f5bce1700a5c49c9:0) is starting
2018-07-05 05:24:01.643 [INF] HSWC: Trimming open circuits for chan_id=530500:1673:0, start_htlc_id=0
2018-07-05 05:24:01.644 [INF] HSWC: Adding live link chan_id=c9495c0a70e1bcf5a018b24fec76bddf32624d9ca4ae64a29e7b82edea84d9e8, short_chan_id=530500:1673:0
2018-07-05 05:24:01.644 [INF] HSWC: HTLC manager for ChannelPoint(e8d984eaed827b9ea264aea49c4d6232dfbd76ec4fb218a0f5bce1700a5c49c9:0) started, bandwidth=0 mSAT
2018-07-05 05:24:01.644 [INF] HSWC: Attempting to re-resynchronize ChannelPoint(e8d984eaed827b9ea264aea49c4d6232dfbd76ec4fb218a0f5bce1700a5c49c9:0)
2018-07-05 05:24:01.644 [INF] CNCT: Attempting to update ContractSignals for ChannelPoint(e8d984eaed827b9ea264aea49c4d6232dfbd76ec4fb218a0f5bce1700a5c49c9:0)
2018-07-05 05:24:01.664 [INF] HSWC: Received re-establishment message from remote side for channel(e8d984eaed827b9ea264aea49c4d6232dfbd76ec4fb218a0f5bce1700a5c49c9:0)
2018-07-05 05:24:01.718 [WRN] HSWC: ChannelPoint(e8d984eaed827b9ea264aea49c4d6232dfbd76ec4fb218a0f5bce1700a5c49c9:0): received unknown message of type *lnwire.ChannelReestablish
2018-07-05 05:24:06.200 [INF] PEER: unable to read message from 188.195.209.57:41810: read tcp 159.89.97.171:9735->188.195.209.57:41810: read: connection reset by peer
```
and
```
2018-07-07 06:47:58.543 [INF] HSWC: HTLC manager for ChannelPoint(0bbb2e3d4b9974c529d6f41f4a66912b2b26024e375bab494ebe704225f9c7d0:0) started, bandwidth=0 mSAT
2018-07-07 06:47:58.543 [INF] HSWC: Attempting to re-resynchronize ChannelPoint(0bbb2e3d4b9974c529d6f41f4a66912b2b26024e375bab494ebe704225f9c7d0:0)
2018-07-07 06:47:58.564 [WRN] HSWC: ChannelPoint(0bbb2e3d4b9974c529d6f41f4a66912b2b26024e375bab494ebe704225f9c7d0:0): received unknown message of type *lnwire.ChannelReestablish
```
It looks like we are starting a channel re-sync, but for some reason we are still getting `ChannelReestablish` messages when we don't expect to. Maybe the c-lightning node is re-sending the `ChannelReestablish` messages unexpectedly?

Or the messages exchanged by the `gossipSyncer` might impact the channel sync somehow... 🤔

**Comment by Roasbeef** [comment]
Looks like `c-lightning` is sending the chan reest message multiple times. If we don't get the initial message, then we'll send an error. My guess is that `c-lightning` thinks that the error is terminal, and force closes the channel as a result. 

Also looking like `c-lighting` accidentally tried to add too many HTLCs to the commitment transaction, and force closed it: 
```
2018-07-03T21:18:32.234Z lightningd(1681): lightning_channeld-02fcc4b0a87749b022f05568d5ef893b7893bcbaa99276bbbb55d9e9abc1bdced0 chan #46: balance = 18446744073573725804, fee is 0, reserve is 5000000
2018-07-03T21:18:32.234Z lightningd(1681): lightning_channeld-02fcc4b0a87749b022f05568d5ef893b7893bcbaa99276bbbb55d9e9abc1bdced0 chan #46: Adding HTLC 0 msat=135825812 cltv=530659 gave CHANNEL_ERR_CHANNEL_CAPACITY_EXCEEDED
2018-07-03T21:18:32.234Z lightningd(1681): lightning_channeld-02fcc4b0a87749b022f05568d5ef893b7893bcbaa99276bbbb55d9e9abc1bdced0 chan #46: REPLY WIRE_CHANNEL_OFFER_HTLC_REPLY with 0 fds
```

**Comment by Roasbeef** [comment]
So in the case that we don't get a reest message, we'll send out an error message. I think it may be the case that c-lightning will _always_ force close on an error message as well.

**Comment by cfromknecht** [comment]
Something seems off here: `balance = 18446744073573725804`, underflow?

Multiple reestablish msgs could also be related to #1761. I suspect the issue may be fixed/improved by #1785 if it stems from the LND side

**Comment by Roasbeef** [comment]
Closing as seems to be a CL issue.
